### DISCUSSION TITLE
Warnings for <a> links

### DISCUSSION THREAD

**Discussion by eyaler** [schemaorg/schemaorg#3385] [Q&A]
Following this [stackoverflow question](https://stackoverflow.com/questions/77279537/how-to-specify-warnings-on-anchor-links), i am interested in marking outgoing links (anchor elements) in an HTML page to specify hazards or other content warnings. My understanding is to do e.g.:
```
<a href="https://www.seizurerobots.com/">
    Killer Japanese Seizure Robots
    <meta itemprop="accessibilityHazard" content="flashing">
</a>

<a href="https://en.wikipedia.org/wiki/Sex">
    Sex
    <meta itemprop="hasAdultConsideration" content="SexualContentConsideration">
</a>
```

Is this the intended way?

This assumes that the added property is understood to relate to the linked content and not to the link element itself (although technically the link element could itself be flashing or contain explicit text)

**Comment by jvandriel** [comment]
> _"This assumes that the added property is understood to relate to the linked content and not to the link element itself..."_

To relate the link (url) to the targeted content you also have to specify what that content is. Meaning your examples need a bit more markup for what you want to express: 

```
<span itemscope itemtype="https://schema.org/WebPage">
  <a itemprop="url" href="https://www.seizurerobots.com/">
      Killer Japanese Seizure Robots
      <meta itemprop="accessibilityHazard" content="flashing">
  </a>
</span>

<span itemscope itemtype="https://schema.org/Product">
  <a itemprop="url" href="https://en.wikipedia.org/wiki/Sex">
      Sex
      <meta itemprop="hasAdultConsideration" content="SexualContentConsideration">
  </a>
</span>
```

You should take note that properties have intended types (and their subtypes) they are supposed to be used with, as mentioned on their pages:

1. [accessibilityHazard](https://schema.org/accessibilityHazard) - Used on these types: Used on these types: [CreativeWork](https://schema.org/CreativeWork)
2. [hasAdultConsideration](https://schema.org/hasAdultConsideration) -  Used on these types: [Offer](https://schema.org/Offer), [Product](https://schema.org/Product)

**Reply by eyaler** [reply]
thanks. in my case NSFW links are not products or offers but webpage/creatework... maybe i should suggest to add these?

**Reply by KalleOlaviNiemitalo** [reply]
Maybe <https://schema.org/contentRating> or <https://schema.org/isFamilyFriendly> could be used.

**Comment by KalleOlaviNiemitalo** [comment]
I'd think you need itemscope attributes as well, and itemprop to indicate that the href attribute value is a property of the item.

So maybe like

```html
<span itemscope itemtype="https://schema.org/WebPage">
<a href="https://www.seizurerobots.com/" itemprop="url">
    Killer Japanese Seizure Robots
    <meta itemprop="accessibilityHazard" content="flashing">
</a>
</span>
```

i.e. there is an item that has that URL and has an accessibility hazard.

This needs the `span` because an `itemscope` attribute on the `a` element would cause `itemprop="url"` to mean that some other item has a "url" property and the value of that property is the item that has the accessibility hazard.  <https://html.spec.whatwg.org/multipage/microdata.html#values>

I also put in an `itemtype` attribute to make it clear that the "url" and "accessibilityHazard" property names refer to the schema.org definitions.

Try it at <https://validator.schema.org/>.

**Reply by KalleOlaviNiemitalo** [reply]
And if you wanted to declare that the link itself has an accessibility hazard, as opposed to the linked web page, I suppose you could do that by declaring an item of type <https://schema.org/WebPageElement> and setting its properties to point to the `a` element.

**Reply by eyaler** [reply]
thanks. why would i need `itemprop="url"`?

**Reply by KalleOlaviNiemitalo** [reply]
Without itemprop, the `href="https://www.seizurerobots.com/"` attribute would be ignored by the algorithm to find the properties of an item <https://html.spec.whatwg.org/multipage/microdata.html#associating-names-with-items>.
